marklogic-client-api/src/test/java/com/marklogic/client/test/StringSearchTest.java

```diff
@@ -107,6 +112,30 @@ public void testStringSearch2() throws IOException {
     }
   }
 
+	@Test
+	public void testStringSearch3() throws IOException {
+		TextDocumentManager textDocumentManager = Common.client.newTextDocumentManager();
+		DocumentWriteSet batch = textDocumentManager.newWriteSet();
+		int numberOfDocs = 100;
+		String[] uris = new String[numberOfDocs];
+		for(int i=0; i<numberOfDocs;i++){
+			uris[i]="/test/forTesting/"+i+".txt";
+			batch.add(uris[i], new StringHandle("Text for doc -"+i).withFormat(Format.TEXT));
+		}
+		textDocumentManager.delete(uris);
+		textDocumentManager.write(batch);
+		QueryManager queryMgr = Common.client.newQueryManager();
+		queryMgr.setPageLength(200);
+		StringQueryDefinition qdef = queryMgr.newStringDefinition();
+		qdef.setDirectory("/test/forTesting/");
+		ArrayList<String> ar = new ArrayList<>();
+		DocumentPage page = textDocumentManager.search(qdef, 1);
+			while(page.hasNext()){
+				DocumentRecord rec = page.next();
+				ar.add(rec.getContent(new StringHandle()).toString());
+			}
+		assertTrue(ar.size()==100,"The number of docs returned is "+ar.size()+" instead of "+numberOfDocs);
```

Comment:
Unrelated note - only use assertTrue/assertFalse when testing a boolean value. If you do `assertEquals(100, ar.size())`, then if it fails, you'll get a nice failure message by default.